arcgis-ios-sdk-samples/Route and directions/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -0,0 +1,151 @@
+        resetBarButtonItem.isEnabled = false
+        // Reset the status text.
+        statusLabel.text = "Satellites info will be shown here."
+        // Reset and stop the location display.
+        mapView.locationDisplay.autoPanModeChangedHandler = nil
+        mapView.locationDisplay.autoPanMode = .off
+        mapView.locationDisplay.stop()
+        // Pause the mock data generation.
+        mockNMEADataSource.stop()
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "DisplayDeviceLocationWithNMEADataSourcesViewController",
+            "SimulatedNMEADataSource"
+        ]
+        // Load NMEA location data source.
+        startBarButtonItem.isEnabled = true
+        nmeaLocationDataSource.locationChangeHandlerDelegate = self
+        mockNMEADataSource.delegate = self
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        reset()
+    }
+}
+
+// MARK: SimulatedNMEADataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: SimulatedNMEADataSourceDelegate {
+    func dataSource(didUpdate nmeaData: Data) {
+        // Push mock data into the data source.
+        // Note: You can also get real-time NMEA sentences from a GPS dongle.
+        nmeaLocationDataSource.push(nmeaData)
+    }
+}
+
+// MARK: AGSNMEALocationDataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: AGSNMEALocationDataSourceDelegate {
+    func nmeaLocationDataSource(_ NMEALocationDataSource: AGSNMEALocationDataSource, satellitesDidChange satellites: [AGSNMEASatelliteInfo]) {
+        // Update the satellites info status text.
+        let satelliteSystemsText = Set(satellites.map { $0.system }).map { $0.label }.joined(separator: ", ")
+        let idText = satellites.map { String($0.satelliteID) }.joined(separator: ", ")
+        let statusText =
+            """
+            \(satellites.count) satellites in view
+            System(s): \(satelliteSystemsText)
+            IDs: \(idText)
+            """
+        statusLabel.text = statusText
+    }
+}
+
+private extension AGSNMEAGNSSSystem {
+    var label: String {
+        switch self {
+        case .GPS:
+            return "The Global Positioning System"
+        case .GLONASS:
+            return "The Russian Global Navigation Satellite System"
+        case .galileo:
+            return "The European Union Global Navigation Satellite System"
+        case .BDS:
+            return "The BeiDou Navigation Satellite System"
+        case .QZSS:
+            return "The Quasi-Zenith Satellite System"
+        case .navIC:
+            return "The Navigation Indian Constellation"
+        default:
+            return "Unknown GNSS type"
```

Comment:
I'm not very familiar with NMEA. The GNSS type didn't change when I tested the sample. In what situation would it change?

Reply:
Since the NMEA sentences are pre-baked, all the satellite information are included in the sentences and all of the satellites were GPS. That is to say, in our sample with the mock data, they won't change.

In real world, it depends on what device to use and where the users are. Maybe some devices in Europe and Asia will use the different satellites such as Galileo, BeiDou, GLONASS, etc. 

I believe at least 90+% chances [people will get GPS](https://forums.quectel.com/t/nmea-active-satellites-and-beidou/6517/2), but I still included all the possible cases in case someone want to dig up the details about our NMEA support in Runtime.

Reply:
Ah I see, thanks for the explanation! I agree, it'd be good to exemplify all cases 👍🏼